### DISCUSSION TITLE
How is the primary repo determined?

### DISCUSSION THREAD

**Discussion by chalin** [cncf/clomonitor#149] [Q&A] [answered]
For example, consider https://clomonitor.io/projects/open-telemetry/open-telemetry, the primary repo is designated as https://github.com/open-telemetry/community. Is this because this is the first pinned repo of the [OpenTelemetry](https://github.com/open-telemetry) org? Thanks. /cc @caniszczyk

**Comment by tegioz** [comment, marked as answer]
Hi @chalin 👋 

We've used the repository listed in the [project page at cncf.io](https://www.cncf.io/projects/opentelemetry/) as the primary repo in most cases. But we can set any other one as primary, it's not a problem 🙂

**Comment by chalin** [comment]
Hi @tegioz! Thanks, that seems fine for now.

**Comment by chalin** [comment]
Actually, come to think of it, for OTel, could you make [open-telemetry/opentelemetry.io](https://github.com/open-telemetry/opentelemetry.io) the primary repo? /cc @austinlparker @nate-double-u

**Reply by austinlparker** [reply]
I don't mind changing it but I don't really have the context for this discussion. What would making the web repo the "main" one do here?

**Reply by chalin** [reply]
The "main" repo get the most analysis (correct me if I'm wrong, @tegioz): for example, see here @austinlparker, https://clomonitor.io/projects/open-telemetry/open-telemetry.

We might want to have a longer discussion here, since OTel doesn't really have a _single_ main code repo, which is why I suggested using the website repo for starters. (But I'll start a separate thread of discussion for that.)

**Reply by tegioz** [reply]
> Could you make [open-telemetry/opentelemetry.io](https://github.com/open-telemetry/opentelemetry.io) the primary repo?

Sure, no problem at all. Before making the change, please let me give you some more details just in case you prefer to handle this in a different way 🙂 

Like you said, the primary repository is analyzed more thoroughly. At the moment it's easier to pass the checks if the primary repository is a *code* repository, as some checks may not apply to, for example, a *docs* repo. Quoting the [checks document](https://github.com/cncf/clomonitor/blob/main/docs/checks.md) we've just added, just in case it helps:

> Many checks rely on checking that certain files exists on a given path. Even though most of these checks support a number of variants, sometimes this won't work for some projects that may be using a different repository layout or organizing content in multiple repositories. In those cases, the recommended approach is to add a section in the README file of the primary repository pointing users to the document location. This will help users discovering this information and will make CLOMonitor happy :) At the moment we support detecting headers as well as links in README files that follow some patterns. Please see the reference below for more information on each case. Some projects have already proceeded this way successfully: https://github.com/kubernetes/kubernetes/pull/108110, https://github.com/kedacore/keda/pull/2704 and https://github.com/cilium/cilium/pull/19037.

We are also [discussing the possibility of supporting more repository kinds](https://github.com/cncf/clomonitor/issues/151), so any feedback is really appreciated.

**Reply by chalin** [reply]
Thanks for the context @tegioz. I'm in favor of supporting more repo kinds, as I mentioned in https://github.com/cncf/clomonitor/issues/151#issuecomment-1067962733. In the meantime, I'd still like to switch OTel's primary repo to refer to the website repo.

**Reply by tegioz** [reply]
No worries. Sure, no problem 👍

**Reply by tegioz** [reply]
Primary repo updated @chalin.
